src/main/java/com/minecolonies/colony/WorkManager.java

```diff
@@ -45,10 +50,14 @@ public WorkManager(Colony c)
      */
     public void addWorkOrder(AbstractWorkOrder order)
     {
+        dirty = true;
+
         if (order.getID() == 0)
         {
             topWorkOrderId++;
+            topPriority++;
```

Comment:
This changes work orders to be FILO instead of FIFO. We don't want new orders to go to the top of the queue, they should be added with priority 0.